examples/advanced_agents/sandbox_agent/src/nat_sandbox_agent/tools/host/web_search.py

```
@@ -0,0 +1,141 @@
+        """
+        self._api_key = api_key or os.environ.get("TAVILY_API_KEY")
+        self._client = None
+
+    def _get_client(self):
+        """Lazily initialize async Tavily client."""
+        if self._client is None:
+            if not self._api_key:
+                raise ValueError(
+                    "TAVILY_API_KEY not set. Please set the environment variable "
+                    "or pass api_key to the tool."
+                )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)